### ISSUE TITLE
Fails to generate code for the gitea API

### ISSUE THREAD

**Issue by dcermak** [MarcoMuellner/openapi-python-generator#7]
**Describe the bug**

openapi-python-generator does not like the API definition of gitea and bugs out during code generation:
```
❯ poetry run openapi-python-generator --library requests https://gitea.opensuse.org/swagger.v1.json gitea/
Generating data from https://gitea.opensuse.org/swagger.v1.json
Error in service adminCronList: invalid syntax (<string>, line 1)
Error in service adminCronRun: invalid syntax (<string>, line 1)
Error in service adminGetAllOrgs: invalid syntax (<string>, line 1)
Error in service adminUnadoptedList: invalid syntax (<string>, line 1)
Error in service adminAdoptRepository: invalid syntax (<string>, line 1)
Error in service adminDeleteUnadoptedRepository: invalid syntax (<string>, line 1)
Error in service adminGetAllUsers: invalid syntax (<string>, line 1)
Error in service adminDeleteUser: invalid syntax (<string>, line 1)
Error in service adminEditUser: invalid syntax (<string>, line 1)
Error in service adminCreatePublicKey: invalid syntax (<string>, line 1)
Error in service adminDeleteUserPublicKey: invalid syntax (<string>, line 1)
Error in service adminCreateOrg: invalid syntax (<string>, line 1)
Error in service adminCreateRepo: invalid syntax (<string>, line 1)
Error in service notifyGetList: invalid syntax (<string>, line 1)
Error in service notifyReadList: invalid syntax (<string>, line 1)
Error in service notifyGetThread: invalid syntax (<string>, line 1)
Error in service notifyReadThread: invalid syntax (<string>, line 1)
Error in service createOrgRepoDeprecated: invalid syntax (<string>, line 1)
Error in service orgGetAll: invalid syntax (<string>, line 1)
Error in service orgGet: invalid syntax (<string>, line 1)
Error in service orgDelete: invalid syntax (<string>, line 1)
Error in service orgEdit: invalid syntax (<string>, line 1)
Error in service orgListHooks: invalid syntax (<string>, line 1)
Error in service orgCreateHook: invalid syntax (<string>, line 1)
Error in service orgGetHook: invalid syntax (<string>, line 1)
Error in service orgDeleteHook: invalid syntax (<string>, line 1)
Error in service orgEditHook: invalid syntax (<string>, line 1)
Error in service orgListLabels: invalid syntax (<string>, line 1)
Error in service orgCreateLabel: invalid syntax (<string>, line 1)
Error in service orgGetLabel: invalid syntax (<string>, line 1)
Error in service orgDeleteLabel: invalid syntax (<string>, line 1)
Error in service orgEditLabel: invalid syntax (<string>, line 1)
Error in service orgListMembers: invalid syntax (<string>, line 1)
Error in service orgIsMember: invalid syntax (<string>, line 1)
Error in service orgDeleteMember: invalid syntax (<string>, line 1)
Error in service orgListPublicMembers: invalid syntax (<string>, line 1)
Error in service orgIsPublicMember: invalid syntax (<string>, line 1)
Error in service orgPublicizeMember: invalid syntax (<string>, line 1)
Error in service orgConcealMember: invalid syntax (<string>, line 1)
Error in service orgListRepos: invalid syntax (<string>, line 1)
Error in service createOrgRepo: invalid syntax (<string>, line 1)
Error in service orgListTeams: invalid syntax (<string>, line 1)
Error in service orgCreateTeam: invalid syntax (<string>, line 1)
Error in service teamSearch: invalid syntax (<string>, line 1)
Error in service listPackages: invalid syntax (<string>, line 1)
Error in service getPackage: invalid syntax (<string>, line 1)
Error in service deletePackage: invalid syntax (<string>, line 1)
Error in service listPackageFiles: invalid syntax (<string>, line 1)
Error in service issueSearchIssues: invalid syntax (<string>, line 1)
Error in service repoSearch: invalid syntax (<string>, line 1)
Error in service repoGet: invalid syntax (<string>, line 1)
Error in service repoDelete: invalid syntax (<string>, line 1)
Error in service repoEdit: invalid syntax (<string>, line 1)
Error in service repoGetArchive: invalid syntax (<string>, line 1)
Error in service repoGetAssignees: invalid syntax (<string>, line 1)
Error in service repoListBranchProtection: invalid syntax (<string>, line 1)
Error in service repoCreateBranchProtection: invalid syntax (<string>, line 1)
Error in service repoGetBranchProtection: invalid syntax (<string>, line 1)
Error in service repoDeleteBranchProtection: invalid syntax (<string>, line 1)
Error in service repoEditBranchProtection: invalid syntax (<string>, line 1)
Error in service repoListBranches: invalid syntax (<string>, line 1)
Error in service repoCreateBranch: invalid syntax (<string>, line 1)
Error in service repoGetBranch: invalid syntax (<string>, line 1)
Error in service repoDeleteBranch: invalid syntax (<string>, line 1)
Error in service repoListCollaborators: invalid syntax (<string>, line 1)
Error in service repoCheckCollaborator: invalid syntax (<string>, line 1)
Error in service repoAddCollaborator: invalid syntax (<string>, line 1)
Error in service repoDeleteCollaborator: invalid syntax (<string>, line 1)
Error in service repoGetRepoPermissions: invalid syntax (<string>, line 1)
Error in service repoGetAllCommits: invalid syntax (<string>, line 1)
Error in service repoGetCombinedStatusByRef: invalid syntax (<string>, line 1)
Error in service repoListStatusesByRef: invalid syntax (<string>, line 1)
Error in service repoGetContentsList: invalid syntax (<string>, line 1)
Error in service repoGetContents: invalid syntax (<string>, line 1)
Error in service repoCreateFile: invalid syntax (<string>, line 1)
Error in service repoUpdateFile: invalid syntax (<string>, line 1)
Error in service repoDeleteFile: invalid syntax (<string>, line 1)
Error in service repoApplyDiffPatch: invalid syntax (<string>, line 1)
Error in service repoGetEditorConfig: invalid syntax (<string>, line 1)
Error in service listForks: invalid syntax (<string>, line 1)
Error in service createFork: invalid syntax (<string>, line 1)
Error in service GetBlob: invalid syntax (<string>, line 1)
Error in service repoGetSingleCommit: invalid syntax (<string>, line 1)
Error in service repoDownloadCommitDiffOrPatch: invalid syntax (<string>, line 1)
Error in service repoGetNote: invalid syntax (<string>, line 1)
Error in service repoListAllGitRefs: invalid syntax (<string>, line 1)
Error in service repoListGitRefs: invalid syntax (<string>, line 1)
Error in service GetAnnotatedTag: invalid syntax (<string>, line 1)
Error in service GetTree: invalid syntax (<string>, line 1)
Error in service repoListHooks: invalid syntax (<string>, line 1)
Error in service repoCreateHook: invalid syntax (<string>, line 1)
Error in service repoListGitHooks: invalid syntax (<string>, line 1)
Error in service repoGetGitHook: invalid syntax (<string>, line 1)
Error in service repoDeleteGitHook: invalid syntax (<string>, line 1)
Error in service repoEditGitHook: invalid syntax (<string>, line 1)
Error in service repoGetHook: invalid syntax (<string>, line 1)
Error in service repoDeleteHook: invalid syntax (<string>, line 1)
Error in service repoEditHook: invalid syntax (<string>, line 1)
Error in service repoTestHook: invalid syntax (<string>, line 1)
Error in service repoGetIssueTemplates: invalid syntax (<string>, line 1)
Error in service issueListIssues: invalid syntax (<string>, line 1)
Error in service issueCreateIssue: invalid syntax (<string>, line 1)
Error in service issueGetRepoComments: invalid syntax (<string>, line 1)
Error in service issueGetComment: invalid syntax (<string>, line 1)
Error in service issueDeleteComment: invalid syntax (<string>, line 1)
Error in service issueEditComment: invalid syntax (<string>, line 1)
Error in service issueGetCommentReactions: invalid syntax (<string>, line 1)
Error in service issuePostCommentReaction: invalid syntax (<string>, line 1)
Error in service issueDeleteCommentReaction: invalid syntax (<string>, line 1)
Error in service issueGetIssue: invalid syntax (<string>, line 1)
Error in service issueDelete: invalid syntax (<string>, line 1)
Error in service issueEditIssue: invalid syntax (<string>, line 1)
Error in service issueGetComments: invalid syntax (<string>, line 1)
Error in service issueCreateComment: invalid syntax (<string>, line 1)
Error in service issueDeleteCommentDeprecated: invalid syntax (<string>, line 1)
Error in service issueEditCommentDeprecated: invalid syntax (<string>, line 1)
Error in service issueEditIssueDeadline: invalid syntax (<string>, line 1)
Error in service issueGetLabels: invalid syntax (<string>, line 1)
Error in service issueAddLabel: invalid syntax (<string>, line 1)
Error in service issueReplaceLabels: invalid syntax (<string>, line 1)
Error in service issueClearLabels: invalid syntax (<string>, line 1)
Error in service issueRemoveLabel: invalid syntax (<string>, line 1)
Error in service issueGetIssueReactions: invalid syntax (<string>, line 1)
Error in service issuePostIssueReaction: invalid syntax (<string>, line 1)
Error in service issueDeleteIssueReaction: invalid syntax (<string>, line 1)
Error in service issueDeleteStopWatch: invalid syntax (<string>, line 1)
Error in service issueStartStopWatch: invalid syntax (<string>, line 1)
Error in service issueStopStopWatch: invalid syntax (<string>, line 1)
Error in service issueSubscriptions: invalid syntax (<string>, line 1)
Error in service issueCheckSubscription: invalid syntax (<string>, line 1)
Error in service issueAddSubscription: invalid syntax (<string>, line 1)
Error in service issueDeleteSubscription: invalid syntax (<string>, line 1)
Error in service issueGetCommentsAndTimeline: invalid syntax (<string>, line 1)
Error in service issueTrackedTimes: invalid syntax (<string>, line 1)
Error in service issueAddTime: invalid syntax (<string>, line 1)
Error in service issueResetTime: invalid syntax (<string>, line 1)
Error in service issueDeleteTime: invalid syntax (<string>, line 1)
Error in service repoListKeys: invalid syntax (<string>, line 1)
Error in service repoCreateKey: invalid syntax (<string>, line 1)
Error in service repoGetKey: invalid syntax (<string>, line 1)
Error in service repoDeleteKey: invalid syntax (<string>, line 1)
Error in service issueListLabels: invalid syntax (<string>, line 1)
Error in service issueCreateLabel: invalid syntax (<string>, line 1)
Error in service issueGetLabel: invalid syntax (<string>, line 1)
Error in service issueDeleteLabel: invalid syntax (<string>, line 1)
Error in service issueEditLabel: invalid syntax (<string>, line 1)
Error in service repoGetLanguages: invalid syntax (<string>, line 1)
Error in service repoGetRawFileOrLFS: invalid syntax (<string>, line 1)
Error in service issueGetMilestonesList: invalid syntax (<string>, line 1)
Error in service issueCreateMilestone: invalid syntax (<string>, line 1)
Error in service issueGetMilestone: invalid syntax (<string>, line 1)
Error in service issueDeleteMilestone: invalid syntax (<string>, line 1)
Error in service issueEditMilestone: invalid syntax (<string>, line 1)
Error in service repoMirrorSync: invalid syntax (<string>, line 1)
Error in service notifyGetRepoList: invalid syntax (<string>, line 1)
Error in service notifyReadRepoList: invalid syntax (<string>, line 1)
Error in service repoListPullRequests: invalid syntax (<string>, line 1)
Error in service repoCreatePullRequest: invalid syntax (<string>, line 1)
Error in service repoGetPullRequest: invalid syntax (<string>, line 1)
Error in service repoEditPullRequest: invalid syntax (<string>, line 1)
Error in service repoDownloadPullDiffOrPatch: invalid syntax (<string>, line 1)
Error in service repoGetPullRequestCommits: invalid syntax (<string>, line 1)
Error in service repoPullRequestIsMerged: invalid syntax (<string>, line 1)
Error in service repoMergePullRequest: invalid syntax (<string>, line 1)
Error in service repoCancelScheduledAutoMerge: invalid syntax (<string>, line 1)
Error in service repoCreatePullReviewRequests: invalid syntax (<string>, line 1)
Error in service repoDeletePullReviewRequests: invalid syntax (<string>, line 1)
Error in service repoListPullReviews: invalid syntax (<string>, line 1)
Error in service repoCreatePullReview: invalid syntax (<string>, line 1)
Error in service repoGetPullReview: invalid syntax (<string>, line 1)
Error in service repoSubmitPullReview: invalid syntax (<string>, line 1)
Error in service repoDeletePullReview: invalid syntax (<string>, line 1)
Error in service repoGetPullReviewComments: invalid syntax (<string>, line 1)
Error in service repoDismissPullReview: invalid syntax (<string>, line 1)
Error in service repoUnDismissPullReview: invalid syntax (<string>, line 1)
Error in service repoUpdatePullRequest: invalid syntax (<string>, line 1)
Error in service repoGetRawFile: invalid syntax (<string>, line 1)
Error in service repoListReleases: invalid syntax (<string>, line 1)
Error in service repoCreateRelease: invalid syntax (<string>, line 1)
Error in service repoGetReleaseByTag: invalid syntax (<string>, line 1)
Error in service repoDeleteReleaseByTag: invalid syntax (<string>, line 1)
Error in service repoGetRelease: invalid syntax (<string>, line 1)
Error in service repoDeleteRelease: invalid syntax (<string>, line 1)
Error in service repoEditRelease: invalid syntax (<string>, line 1)
Error in service repoListReleaseAttachments: invalid syntax (<string>, line 1)
Error in service repoCreateReleaseAttachment: invalid syntax (<string>, line 1)
Error in service repoGetReleaseAttachment: invalid syntax (<string>, line 1)
Error in service repoDeleteReleaseAttachment: invalid syntax (<string>, line 1)
Error in service repoEditReleaseAttachment: invalid syntax (<string>, line 1)
Error in service repoGetReviewers: invalid syntax (<string>, line 1)
Error in service repoSigningKey: invalid syntax (<string>, line 1)
Error in service repoListStargazers: invalid syntax (<string>, line 1)
Error in service repoListStatuses: invalid syntax (<string>, line 1)
Error in service repoCreateStatus: invalid syntax (<string>, line 1)
Error in service repoListSubscribers: invalid syntax (<string>, line 1)
Error in service userCurrentCheckSubscription: invalid syntax (<string>, line 1)
Error in service userCurrentPutSubscription: invalid syntax (<string>, line 1)
Error in service userCurrentDeleteSubscription: invalid syntax (<string>, line 1)
Error in service repoListTags: invalid syntax (<string>, line 1)
Error in service repoCreateTag: invalid syntax (<string>, line 1)
Error in service repoGetTag: invalid syntax (<string>, line 1)
Error in service repoDeleteTag: invalid syntax (<string>, line 1)
Error in service repoListTeams: invalid syntax (<string>, line 1)
Error in service repoCheckTeam: invalid syntax (<string>, line 1)
Error in service repoAddTeam: invalid syntax (<string>, line 1)
Error in service repoDeleteTeam: invalid syntax (<string>, line 1)
Error in service repoTrackedTimes: invalid syntax (<string>, line 1)
Error in service userTrackedTimes: invalid syntax (<string>, line 1)
Error in service repoListTopics: invalid syntax (<string>, line 1)
Error in service repoUpdateTopics: invalid syntax (<string>, line 1)
Error in service repoAddTopic: invalid syntax (<string>, line 1)
Error in service repoDeleteTopic: invalid syntax (<string>, line 1)
Error in service repoTransfer: invalid syntax (<string>, line 1)
Error in service acceptRepoTransfer: invalid syntax (<string>, line 1)
Error in service rejectRepoTransfer: invalid syntax (<string>, line 1)
Error in service repoCreateWikiPage: invalid syntax (<string>, line 1)
Error in service repoGetWikiPage: invalid syntax (<string>, line 1)
Error in service repoDeleteWikiPage: invalid syntax (<string>, line 1)
Error in service repoEditWikiPage: invalid syntax (<string>, line 1)
Error in service repoGetWikiPages: invalid syntax (<string>, line 1)
Error in service repoGetWikiPageRevisions: invalid syntax (<string>, line 1)
Error in service generateRepo: invalid syntax (<string>, line 1)
Error in service repoGetByID: invalid syntax (<string>, line 1)
Error in service orgGetTeam: invalid syntax (<string>, line 1)
Error in service orgDeleteTeam: invalid syntax (<string>, line 1)
Error in service orgEditTeam: invalid syntax (<string>, line 1)
Error in service orgListTeamMembers: invalid syntax (<string>, line 1)
Error in service orgListTeamMember: invalid syntax (<string>, line 1)
Error in service orgAddTeamMember: invalid syntax (<string>, line 1)
Error in service orgRemoveTeamMember: invalid syntax (<string>, line 1)
Error in service orgListTeamRepos: invalid syntax (<string>, line 1)
Error in service orgListTeamRepo: invalid syntax (<string>, line 1)
Error in service orgAddTeamRepository: invalid syntax (<string>, line 1)
Error in service orgRemoveTeamRepository: invalid syntax (<string>, line 1)
Error in service topicSearch: invalid syntax (<string>, line 1)
Error in service userGetOauth2Application: invalid syntax (<string>, line 1)
Error in service userGetOAuth2Application: invalid syntax (<string>, line 1)
Error in service userDeleteOAuth2Application: invalid syntax (<string>, line 1)
Error in service userUpdateOAuth2Application: invalid syntax (<string>, line 1)
Error in service userCurrentListFollowers: invalid syntax (<string>, line 1)
Error in service userCurrentListFollowing: invalid syntax (<string>, line 1)
Error in service userCurrentCheckFollowing: invalid syntax (<string>, line 1)
Error in service userCurrentPutFollow: invalid syntax (<string>, line 1)
Error in service userCurrentDeleteFollow: invalid syntax (<string>, line 1)
Error in service userCurrentListGPGKeys: invalid syntax (<string>, line 1)
Error in service userCurrentGetGPGKey: invalid syntax (<string>, line 1)
Error in service userCurrentDeleteGPGKey: invalid syntax (<string>, line 1)
Error in service userCurrentListKeys: invalid syntax (<string>, line 1)
Error in service userCurrentGetKey: invalid syntax (<string>, line 1)
Error in service userCurrentDeleteKey: invalid syntax (<string>, line 1)
Error in service orgListCurrentUserOrgs: invalid syntax (<string>, line 1)
Error in service userCurrentListRepos: invalid syntax (<string>, line 1)
Error in service userCurrentListStarred: invalid syntax (<string>, line 1)
Error in service userCurrentCheckStarring: invalid syntax (<string>, line 1)
Error in service userCurrentPutStar: invalid syntax (<string>, line 1)
Error in service userCurrentDeleteStar: invalid syntax (<string>, line 1)
Error in service userGetStopWatches: invalid syntax (<string>, line 1)
Error in service userCurrentListSubscriptions: invalid syntax (<string>, line 1)
Error in service userListTeams: invalid syntax (<string>, line 1)
Error in service userCurrentTrackedTimes: invalid syntax (<string>, line 1)
Error in service userSearch: invalid syntax (<string>, line 1)
Error in service userGet: invalid syntax (<string>, line 1)
Error in service userListFollowers: invalid syntax (<string>, line 1)
Error in service userListFollowing: invalid syntax (<string>, line 1)
Error in service userCheckFollowing: invalid syntax (<string>, line 1)
Error in service userListGPGKeys: invalid syntax (<string>, line 1)
Error in service userGetHeatmapData: invalid syntax (<string>, line 1)
Error in service userListKeys: invalid syntax (<string>, line 1)
Error in service orgListUserOrgs: invalid syntax (<string>, line 1)
Error in service orgGetUserPermissions: invalid syntax (<string>, line 1)
Error in service userListRepos: invalid syntax (<string>, line 1)
Error in service userListStarred: invalid syntax (<string>, line 1)
Error in service userListSubscriptions: invalid syntax (<string>, line 1)
Error in service userGetTokens: invalid syntax (<string>, line 1)
Error in service userCreateToken: invalid syntax (<string>, line 1)
Error in service userDeleteAccessToken: invalid syntax (<string>, line 1)
Traceback (most recent call last):
  File "/home/dan/.cache/pypoetry/virtualenvs/scm-staging-UcbLi63M-py3.11/bin/openapi-python-generator", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/dan/.cache/pypoetry/virtualenvs/scm-staging-UcbLi63M-py3.11/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dan/.cache/pypoetry/virtualenvs/scm-staging-UcbLi63M-py3.11/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/home/dan/.cache/pypoetry/virtualenvs/scm-staging-UcbLi63M-py3.11/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dan/.cache/pypoetry/virtualenvs/scm-staging-UcbLi63M-py3.11/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dan/.cache/pypoetry/virtualenvs/scm-staging-UcbLi63M-py3.11/lib/python3.11/site-packages/openapi_python_generator/__main__.py", line 49, in main
    generate_data(source, output, library, env_token_name, use_orjson)
  File "/home/dan/.cache/pypoetry/virtualenvs/scm-staging-UcbLi63M-py3.11/lib/python3.11/site-packages/openapi_python_generator/generate_data.py", line 143, in generate_data
    write_data(result, output)
  File "/home/dan/.cache/pypoetry/virtualenvs/scm-staging-UcbLi63M-py3.11/lib/python3.11/site-packages/openapi_python_generator/generate_data.py", line 112, in write_data
    write_code(
  File "/home/dan/.cache/pypoetry/virtualenvs/scm-staging-UcbLi63M-py3.11/lib/python3.11/site-packages/openapi_python_generator/generate_data.py", line 33, in write_code
    formatted_contend = black.format_file_contents(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "src/black/__init__.py", line 926, in format_file_contents
  File "src/black/__init__.py", line 1073, in format_str
  File "src/black/__init__.py", line 1083, in _format_str_once
  File "src/black/parsing.py", line 127, in lib2to3_parse
black.parsing.InvalidInput: Cannot parse: 8:28: def createOrgRepoDeprecated(, body : CreateRepoOption = None, ) -> Repository:
```
I have tried using a different library (and ran into #6), but that does not appear to help.

**Comment by MarcoMuellner** [comment]
I'll take a look. For posterities sake, this is the [openapi.json](https://gitea.opensuse.org/swagger.v1.json).

**Comment by MarcoMuellner** [comment]
According to the provided openapi.json, it is currently compliant with openapi v2, which this project does not support as stated in the [docs](https://marcomuellner.github.io/openapi-python-generator/).

This could maybe be a thing for the future if the interest is high enough, however as of now, this is no priority for me. I would be glad for a pull request if you want to make one :)

**Comment by MarcoMuellner** [comment]
This should be fixed with the latest [v0.3.10](https://github.com/MarcoMuellner/openapi-python-generator/releases/tag/v0.3.10) release of the tool :)

**Comment by dcermak** [comment]
Yes, this works now, thank you!!
